App.js

```diff
@@ -1,141 +1,126 @@
+      return require("./assets/cloudy.jpg");
     } else if (temperature > 15 && temperature <= 25) {
-      return require('./assets/sunny.jpeg');
+      return require("./assets/sunny.jpeg");
     } else {
-      return require('./assets/desert.jpg');
+      return require("./assets/desert.jpg");
     }
   };
 
-  
   return (
-    <View style={styles.root}>
+    <View style={styles.container}>
       <ImageBackground
         source={getBackgroundImage(data?.main?.temp)}
         resizeMode="cover"
-        style={styles.image}>
-
-        {data &&
+        style={styles.image}
+      >
+        <SearchInput
+          input={input}
+          setInput={setInput}
+          onSubmitEditing={fetchDataWeather}
+        />
+
+        {data && (
           <View style={styles.infoView}>
             <Text style={styles.cityCountyText}>
               {`${data?.name}, ${data?.sys?.country}`}
             </Text>
-            <Text style={styles.displayTemp}>{`${Math.round(data?.main?.temp,)}°C`}</Text>
-            <Text style={styles.minMaxTemp}>{`Min ${Math.round(data?.main?.temp_min,
-            )}°C / Max ${Math.round(data?.main?.temp_max,)}°C`}</Text>
-          </View>}
+            <Text style={styles.displayTemp}>{`${Math.round(
+              data?.main?.temp
+            )}°C`}</Text>
+            <Text style={styles.minMaxTemp}>{`Min ${Math.round(
+              data?.main?.temp_min
+            )}°C / Max ${Math.round(data?.main?.temp_max)}°C`}</Text>
+          </View>
+        )}
 
-        <View>
-          <TextInput
-            placeholder="Enter city name..."
-            onChangeText={text => setInput(text)}
-            value={input}
-            placeholderTextColor={'#fff'}
-            style={styles.textInput}
-            onSubmitEditing={fetchDataWeather}
-          />
-        </View>
-        {loading &&
+        {loading && (
           <View>
-            <ActivityIndicator size={'large'} color='#000' />
+            <ActivityIndicator size={"large"} color="#000" />
           </View>
-        }
-
+        )}
       </ImageBackground>
     </View>
   );
 }
 
 const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    backgroundColor: '#fff',
-    alignItems: 'center',
-    justifyContent: 'center',
-  },
-
-  image: {
-    flex: 1,
-    flexDirection: 'column',
-  },
-
-  root: {
-    flex: 1,
-  },
-
-  textInput: {
-    borderBottomWidth: 3,
-    padding: 5,
-    paddingVertical: 10,
-    marginVertical: 50,
-    marginHorizontal: 10,
-    backgroundColor: '#696969',
-    color: "#fff",
-    fontSize: 19,
-    borderRadius: 16,
-    borderBottomColor: 'coral',
-  },
-
-  infoView: {
-    alignItems: 'center',
-  },
-
   cityCountyText: {
-    color: '#fff',
+    color: "#fff",
     fontSize: 40,
-    fontWeight: 'bold',
+    fontWeight: "bold",
     marginTop: 40,
   },
-
+  container: {
+    flex: 1,
+    backgroundColor: "#fff",
+    justifyContent: "center",
```

Comment:
In contextul modificarilor actuale poti sa lasi `alignItems: "center"` in `container`